### ISSUE TITLE
Span-based NER training throws IndexOutOfRange in CrossEntropyWithLogitsLoss one-hot path

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1741]
## Summary

Once the MultiHeadAttention dimension mismatch (#1725, fixed in #1740) is
resolved, span-based NER models (`SpERTNER` and likely the rest of the
`SpanBasedNERBase` family — `BiaffineNER`, `PURENER`, `PyramidNER`, `TriaffineNER`,
`W2NER`) fail their **training** Generated tests with:

```
System.IndexOutOfRangeException : Index was outside the bounds of the array.
  at AiDotNet.LossFunctions.CrossEntropyWithLogitsLoss`1.ClassIndicesToOneHot(...) :line 353
  at AiDotNet.LossFunctions.CrossEntropyWithLogitsLoss`1.ComputeTapeLoss(...) :line 152
  at AiDotNet.NeuralNetworks.NeuralNetworkBase`1.TrainWithTape(...) :line 7260
  at AiDotNet.NER.SpanBased.SpanBasedNERBase`1.Train(...) :line 295
```

`SpERTNERTests`: **15 pass / 12 fail** — every failure is this exception in a
training path (`GradientFlow_ShouldBeNonZeroAndFinite`,
`DifferentInputs_AfterTraining`, `MoreData_ShouldNotDegrade`, etc.). The
non-training (forward/metadata/determinism) tests pass.

## Root cause (preliminary)

`ClassIndicesToOneHot` enters the class-index branch because
`target.Shape.Length == predicted.Shape.Length - 1`. The test target is shaped
from `EffectiveOutputShape`, which is inferred from a warm-up `Predict()` — but
`Predict()` returns **decoded class indices** while the training-time forward
emits **per-token logits** `[seqLen, numLabels]`. The one-hot offset mapping in
`ClassIndicesToOneHot` (line ~349-353) then computes a flat offset that can run
past the one-hot buffer when the indices' implied non-class extents don't line up
with the logits' non-class axes, producing the IndexOutOfRange.

This is a **train-time target/prediction shape-contract bug**, distinct from the
#1725 MHA dimension class.

## Suggested direction

1. Make `SpanBasedNERBase.Train` (or the span-NER predict/forward contract) align
   the training target shape with the logits the loss actually compares against,
   rather than relying on `Predict()`'s decoded shape.
2. Harden `CrossEntropyWithLogitsLoss.ClassIndicesToOneHot` to validate that the
   class-index target's non-class extents match the logits' non-class axes and
   throw a clear shape-contract error (not a raw IndexOutOfRange) — without
   silently dropping labels.

## Acceptance
- `Generated.SpERTNERTests` training tests pass (or are tagged HeavyTimeout if
  genuinely foundation-scale after the fix).
- Same for the other `SpanBasedNERBase` family models that share this path.

Follow-up to #1725 / #1740.
